jdaviz/configs/imviz/plugins/footprints/footprints.py

```diff
@@ -185,6 +185,12 @@ def _on_link_type_updated(self, msg=None):
         # toggle visibility as necessary
         self._on_is_active_changed()
 
+        # When fp(s) added via API before WCS link. Overlays visibility & is_active
+        # can be True, but only last fp will display. This ensures all fp(s) display
+        if not self.is_pixel_linked:
+            for choice in self.overlay.choices:
+                self.overlay.selected = choice

```

Comment:
can we call the method internally so that the selected choice isn't affected (I'm assuming this works because it triggers the observe on `overlay_selected`)?  If that isn't possible, can we at least restore the user-selection after this for-loop?

Reply:
It is triggering the observe. I am seeing that without changing the selected choice and calling the method internally, the only overlay that displays is the last one added based on `preset_args_changed`.

Reply:
can we have the internal method take the name of the overlay and _default_ to overlay_selected so that we can trigger the logic without changing the selection?

Reply:
see https://github.com/gibsongreen/jdaviz/pull/1 for a rough idea of how this could work (although I didn't have a test case, so haven't tested if it still fixes the bug - a regression test would be great if possible!)